src/main/java/com/kainos/discoverydiary/resources/MediaResource.java

```diff
@@ -23,4 +25,19 @@ public MediaResource() {
     public View Detail(@PathParam("id") int id) {
         return new Detail(DataStore.medias.get(id));
     }
+
+    @POST
+    @Timed
+    @Path("{id}")
+    public View Borrow(@PathParam("id") int id, @FormParam("nameOfBorrower") String nameOfBorrower) {
+
+        Media media = DataStore.medias.get(id);
+
+        if (media.getStatus() == Status.AVAILABLE) {
+            media.setNameOfBorrower(nameOfBorrower);
+            media.setStatus(Status.ON_LOAN);
+        }
+
+        return Detail(id);
```

Comment:
redirect here, might need to change return type to "Response" and set the entity to "Detail(id)"